### PR TITLE
chat main

### PR DESCRIPTION
**Pull request by fire-poy**
FINALLY!!!

The first version of chat 100% functional (?)

It's not perfect but it's good enough so we can test it in PROD and see all the problems that we don't see alone.

If you don't understand, don't worry it's normal, I can answer your questions in the reunion.

Thanks to @RobinBurri for his enormous help!, he made almost half of the work I think.

### CODE REVIEW

File: backend/src/user/user.controller.ts [hunk 1]
```diff
@@ -123,6 +124,17 @@ export class UserController {
...
+        const user = await this.userService.findOne(req.user.id)
+        if (!user) {
+            throw new NotFoundException('User not found')
+        }
+        const { TFASecret, FT_id, ...rest } = user
+        const userPosition = await this.userService.getUserRankingPosition(
+            req.user.id
+        )
+        return { ...rest, userPosition }
+    }
```
Comment (eprei) [inline backend/src/user/user.controller.ts:137]: can we delete this?

Comment (eprei) [inline backend/src/user/user.controller.ts:137]: ``
```suggestion
```
Can we delete this? I think this is a code left over from some rebase maybe.

**Comment by eprei** [comment]
Wow, huge job @fire-poy and @RobinBurri . 

Congratulations 🎉

File: backend/src/user/user.service.ts [hunk 2]
```diff
@@ -238,7 +238,7 @@ export class UserService {
             throw new NotFoundException('User not found')
         }
 
-        const { id, TFASecret, FT_id, ...rest } = user
+        const { TFASecret, FT_id, ...rest } = user
```
Comment (eprei) [inline backend/src/user/user.service.ts:241]: This is so that the id of user 'me' is available in the store and can be used in the frontend, right?

Comment (RobinBurri) [reply backend/src/user/user.service.ts:241]: yes

File: backend/src/user/user.controller.ts (hunk 1 above)
Comment (RobinBurri) [reply backend/src/user/user.controller.ts]: yes you are right

**Comment by eprei** [comment]
Great job, bravo! I know you have been working hard on this feature. I have verified that it works globally and that it does not conflict with parts I have developed. So as long as it is ready to merge for you, it will be ready to merge for me. I'll leave you with my early validation.
Once again, bravo!

**Review by shaolin-peanut — APPROVED** [review]
Impressive work! Congrats and thank you for all this effort 🙏

**Review by shaolin-peanut — COMMENTED** [review]
I observed something. When you start a dm with yourself and send some messages, then change the name of yourself and send more messages, it seems like there are two people talking to each other. I don't know if it's a big issue, at least you know it happens
image

**Pull request merged**